### ISSUE TITLE
Product attribute option manager's existing option check fails

### ISSUE THREAD

**Issue by jarnooravainen** [magento/magento2#18246]
### Preconditions
1. Magento Commerce 2.2.5 (version of module-catalog is 102.0.5)

### Steps to reproduce
1. Add attribute options by calling \Magento\Catalog\Model\Product\Attribute\OptionManagement::add
1. Add attribute option that has admin scope value '01'.
2. Add another option for same attribute that has admin scope value '1'.

### Expected result
1. Both options can be found from table eav_attribute_option_value.

### Actual result
1. Only option with value '01' exists in DB.

### Origin of the issue
https://github.com/magento/magento2/blob/547ad4a31f135aca46e8620b8f59e55b83a877a6/app/code/Magento/Catalog/Model/Product/Attribute/OptionManagement.php#L50

In_Array comparison fails due to fact that comparison is not done "strictly".

### Suggested fix
`if (in_array((string)$option->getLabel(), $currentOptions, true)) {`

**Comment by magento-engcom-team** [comment]
Hi @jarnooravainen. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@jarnooravainen do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @jarnooravainen thank you for you report, the fix for this issue has already available in 2.3-develop branch

**Comment by Ctucker9233** [comment]
@engcom-backlog-nazar This issue was reported for 2.2 not 2.3. Shouldn't this issue remain open until it's fixed in the branch it was reported against? If a backport can be created, link commits and I'd be happy to take this.

**Comment by ihor-sviziev** [comment]
Hi @jarnooravainen. Thank you for your report. 
The issue has been fixed in magento/magento2#19612 by @SikailoISM in 2.2-develop branch
Related commit(s):
 - [14aa091388aaea80195d874e47d874f8df6620d7](https://github.com/magento/magento2/commit/14aa091388aaea80195d874e47d874f8df6620d7)

The fix will be available with the upcoming 2.2.8 release.

**Comment by ihor-sviziev** [comment]
Duplicate of #19436
